kyuubi-server/src/main/scala/org/apache/kyuubi/server/mysql/MySQLRichByteBuf.scala

```diff
@@ -0,0 +1,320 @@
+     *
+     * @param value 2 byte fixed length integer
+     */
+    def writeInt2(value: Int): ByteBuf = self.writeShortLE(value)
+
+    /**
+     * Read 3 byte fixed length integer from byte buffers.
+     *
+     * @return 3 byte fixed length integer
+     */
+    def readInt3: Int = self.readUnsignedMediumLE
+
+    /**
+     * Write 3 byte fixed length integer to byte buffers.
+     *
+     * @param value 3 byte fixed length integer
+     */
+    def writeInt3(value: Int): ByteBuf = self.writeMediumLE(value)
+
+    /**
+     * Read 4 byte fixed length integer from byte buffers.
+     *
+     * @return 4 byte fixed length integer
+     */
+    def readInt4: Int = self.readIntLE
+
+    /**
+     * Write 4 byte fixed length integer to byte buffers.
+     *
+     * @param value 4 byte fixed length integer
+     */
+    def writeInt4(value: Int): ByteBuf = self.writeIntLE(value)
+
+    /**
+     * Read 6 byte fixed length integer from byte buffers.
+     *
+     * @return 6 byte fixed length integer
+     */
+    def readInt6: Long = (0 until 6).foldLeft(0L) { case (result, i) =>
+      result | self.readUnsignedByte.toLong << (8 * i)
+    }
+
+    /**
+     * Write 6 byte fixed length integer to byte buffers.
+     *
+     * @param value 6 byte fixed length integer
+     */
+    def writeInt6(value: Long): ByteBuf = throw new UnsupportedOperationException
+
+    /**
+     * Read 8 byte fixed length integer from byte buffers.
+     *
+     * @return 8 byte fixed length integer
+     */
+    def readInt8: Long = self.readLongLE
+
+    /**
+     * Write 8 byte fixed length integer to byte buffers.
+     *
+     * @param value 8 byte fixed length integer
+     */
+    def writeInt8(value: Long): ByteBuf = self.writeLongLE(value)
+
+    /**
+     * Read lenenc integer from byte buffers.
+     *
+     * @return lenenc integer
+     */
+    def readIntLenenc: Long = {
+      val firstByte = readInt1
+      if (firstByte < 0xfb) return firstByte
+      if (0xfb == firstByte) return 0
+      if (0xfc == firstByte) return readInt2
+      if (0xfd == firstByte) return readInt3
+      self.readLongLE
+    }
+
+    /**
+     * Write lenenc integer to byte buffers.
+     *
+     * @param value lenenc integer
+     */
+    def writeIntLenenc(value: Long): ByteBuf = {
+      if (value < 0xfb) {
+        self.writeByte(value.toInt)
+      } else if (value < Math.pow(2, 16)) {
+        self.writeByte(0xfc)
+        self.writeShortLE(value.toInt)
+      } else if (value < Math.pow(2, 24)) {
+        self.writeByte(0xfd)
+        self.writeMediumLE(value.toInt)
+      } else {
+        self.writeByte(0xfe)
+        self.writeLongLE(value)
+      }
+    }
+
+    /**
+     * Read fixed length long from byte buffers.
+     *
+     * @param length length read from byte buffers
+     * @return fixed length long
+     */
+    def readLong(length: Int): Long =
+      (0 until length).foldLeft(0) { case (result, _) => result << 8 | readInt1 }
```

Comment:
can we use `while` ?

I think the result should be defined as `long` ?

Reply:
Scala `Long` is Java `long`